compiler/src/dotty/tools/dotc/core/GadtConstraint.scala

```diff
@@ -262,27 +255,28 @@ final class ProperGadtConstraint private(
 
   // ---- Private ----------------------------------------------------------
 
-  private def externalize(param: TypeParamRef)(using Context): Type =
-    reverseMapping(param) match {
+  private def externalize(tp: Type, theMap: TypeMap | Null = null)(using Context): Type = tp match
+    case param: TypeParamRef => reverseMapping(param) match
       case sym: Symbol => sym.typeRef
-      case null => param
-    }
+      case null        => param
+    case tp: TypeAlias       => tp.derivedAlias(externalize(tp.alias))
```

Comment:
Shall we pass `theMap` to the recursive call to `externalize` here?

Reply:
Wow, nice eagle eye!  Yes!